app/src/components/BaseImageUpload.vue

```diff
@@ -1,16 +1,67 @@
 <template>
-  <div :class="width">
-    <label class="flex flex-col bg-white items-center block px-6 py-8 border border-grey-400 cursor-pointer">
-      <span>Choose image file to upload</span>
-      <div v-if="isValid && val" :id="`${id}-success`">Uploaded: {{ val.name }}</div>
-      <input type="file" @input="onInput" class="hidden" />
-    </label>
-    <div v-if="!isValid" class="bg-pink p-4 text-white" :id="`${id}-error`">{{ errorMsg }}</div>
+  <!-- Image Upload -->
+  <template v-if="!logoURI || !isValid">
+    <div class="block">
+      <label class="w-100 flex flex-col bg-white items-center px-6 py-8 border border-grey-400 cursor-pointer">
+        <input type="file" @input="onInput" class="hidden" />
+        <div class="flex p-12 gap-x-12">
+          <div
+            class="
+              group
+              flex
+              items-center
+              justify-center
+              h-24
+              border border-grey-400
+              p-12
+              cursor-pointer
+              hover:border-grey-500
+            "
+          >
+            <ImportIcon class="icon icon-small icon-primary" />
+          </div>
+          <div class="text-grey-400 flex items-center">
+            1920x1080px<br />
+            *.png
+          </div>
+        </div>
+      </label>
+    </div>
+
+    <div v-if="logoURI && !isValid" class="bg-pink p-4 text-white" :id="`${id}-error`">
+      {{ errorMsg }}
+    </div>
+  </template>
+
+  <template v-else-if="isUploading">
+    <span>Uploading to IPFS...</span>
+  </template>
+
+  <!-- Uploaded Image Preview -->
+  <div v-else-if="logoURI" class="py-8">
```

Comment:
How did you decided to use `<template>` for the first two `if` conditions, and `<div>` for the third? Curious if there's a best practice here we're following. The only difference I'm aware of is that `<template>` doesn't render in the DOM, so in this component they seem interchangeable

Reply:
- so the first one has 2 divs and it seemed redundant enclosing a div and hence used a template
- the second one which has Uploading to IPFS doesn't need a div and hence used a template (the argument could be made i just apply the if on the span tag
- the last one, there is an enclosing div so i ended up adding it to that instead of wrapping a template.

I'm not sure what the cleanest way here is but open to suggestions

Reply:
I don't feel strongly about it, was just curious! I think the guidance is "do you need a div or does a div make sense, if yes `div`, if no `template`"